eZ/Publish/Core/MVC/Symfony/FieldType/Relation/ParameterProvider.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace eZ\Publish\Core\MVC\Symfony\FieldType\Relation;
+
+use eZ\Publish\API\Repository\ContentService;
+use eZ\Publish\API\Repository\Values\Content\Field;
+use eZ\Publish\Core\MVC\Symfony\FieldType\View\ParameterProviderInterface;
+
+class ParameterProvider implements ParameterProviderInterface
+{
+    /** @var \eZ\Publish\API\Repository\ContentService */
+    private $contentService;
+
+    /**
+     * @param \eZ\Publish\API\Repository\ContentService $contentService
+     */
+    public function __construct(ContentService $contentService)
+    {
+        $this->contentService = $contentService;
+    }
+
+    /**
+     * Returns a hash of parameters to inject to the associated fieldtype's view template.
+     * Returned parameters will only be available for associated field type.
+     *
+     * Key is the parameter name (the variable name exposed in the template, in the 'parameters' array).
+     * Value is the parameter's value.
+     *
+     * @param \eZ\Publish\API\Repository\Values\Content\Field $field The field parameters are provided for.
+     *
+     * @return array
+     */
+    public function getViewParameters(Field $field)
+    {
+        $contentInfo = $this->contentService->loadContentInfo($field->value->destinationContentId);
```

Comment:
AFAIK `destinationContentId` will point to non-existing content id after empty the trash which means we will still have that same issue. We should catch `NotFoundException` here to avoid this (in same way as we solved similar issue in RichText)

Reply:
If so, we have yet another issue (partially related to this one): reverse relations are not removed along with the content. @andrerom ?

Reply:
Or as mentioned on slack, use SPI / or [expose in API](https://github.com/ezsystems/ezpublish-kernel/pull/2070) SPI's ContentHandler->loadContentInfoList().

It won't throw on missing items so you won't have to catch anything. But if we opt to use SPI we will also need to check permissions and filter on that, but otherwise will allow us to load all relations at once for better performance on this. Long term this should be exposed in API, Page Builder needs something like this for loading all content / content info for misc blocks.

Reply:
@Nattfarinn Correct, we have left it up to view layer to filter it out, afaik should be logic for that in some cases already like RichText.

Theory on this is that we need editor to figure out what to do with it, especially  in case of richtext embed, so we don't want to silently remove them. Also besides such cleanup making lots of assumptions on what we think is best, it will also be costly to update field data on deletion.

So there are a few cases to try to filter on in view:
- deleted?
- trashed?
- hidden?
- user has access?

If we continue down this road, a better define on how field types should handle this is missing, across relation(list)/richtext/page-builder, including concept of differing between raw editorial data vs filtered, and expose info to UI on why items gets filtered out in order to tell editor "Before you are allowed to publish this draft, please change relation/embed on field X which does no longer exist." Same goes for page builder.

Reply:
> deleted?

I think this case is a bit different from the others, because we don't have `Content` object anymore in database. The `destinationContentId` will never lead to a proper content and it's irreversible. Isn't it something we should have `SignalSlot` for? `DeleteContentSignal` is emitted, and so we could do something like:

```php
public function receive(Signal $signal)
{
    $reverseRelations = $this->persistenceHandler->contentHandler()->loadReverseRelations($signal->contentId);

    foreach ($reverseRelations as $relation) {
        $this->persistenceHandler->contentHandler()->removeRelation($relation->id, $relation->type);
    }
}
```

At least in this case.

Reply:
How should it handle updated field types? call all field types relevant to the content and tell them to clean themselves up? Should we do that on subtree delete? How should it del with case where for instance field is required? Or maybe other logic meaning editor should handle it.

It's possible, but it will have it's own set of issues.